src/unwind.cc

```diff
@@ -123,11 +123,16 @@ DDRes unwindstate_unwind(UnwindState *us) {
   return res;
 }
 
-void unwind_pid_free(UnwindState *us, pid_t pid) {
-  us->dso_hdr.pid_free(pid);
-  us->dwfl_hdr.clear_pid(pid);
-  us->symbol_hdr.clear(pid);
-  us->process_hdr.clear(pid);
+void unwind_pid_free(UnwindState *us, pid_t pid,
+                     PerfClock::time_point timestamp) {
+  if (!(us->dso_hdr.pid_free(pid, timestamp))) {
+    LG_DBG("(PID Free)%d -> avoid free of mappings (%ld)", pid,
+           timestamp.time_since_epoch().count());
+  } else {
```

Comment:
if the mappings are good. Does it really mean that the underlying dwfl modules are OK ?
Today we do not really tie the lifetime of the mappings to the lifetimes of the unwinding caches.
So I am not sure about this change. I think it requires more careful thinking.
@nsavoire fyi

Reply:
I agree